File: python/examples/entities_linked.py
# -*- coding: utf-8 -*-

"""
Example code to call Rosette API to get linked (against Wikipedia) entities from a piece of text.
"""

import json

from rosette.api import API, DocumentParameters

def run(key):
    # Create an API instance
    api = API(user_key = key)

    params = DocumentParameters()
    params["content"] = u"${entities_linked_data}"
    result = api.entities(params, True)  # entity linking is turned on

    print(json.dumps(result, indent=2, ensure_ascii=False).encode("utf8"))
    return json.dumps(result, indent=2, ensure_ascii=False).encode("utf8")

File: nodejs/examples/entities_linked.js
/*
 * Example code to call Rosette API to get linked (against Wikipedia) entities from a piece of text.
 */

"use strict";

var Api = require("rosette-api").Api;
var ArgumentParser = require("argparse").ArgumentParser;
var DocumentParameters = require("rosette-api").DocumentParameters;

var parser = new ArgumentParser({
  addHelp: true,
  description: "Get linked entities from a piece of text"
});
parser.addArgument(["--key"], {help: "Rosette API key", required: true});
var args = parser.parseArgs();

var docParams = new DocumentParameters();
var content = "${entities_linked_data}";
docParams.setItem("content", content);

var api = new Api(args.key);
api.entities(docParams, true, function(err, res) {
  if (err) {
    throw err;
  }
  else {
    console.log(res);
  }
});
